traffic_portal/test/end_to_end/common/commonFunctions.js

```diff
@@ -31,4 +31,17 @@ module.exports = function() {
 	this.urlPath = function ( url ) {
 		return '/' + String(url).split('/').slice(3).join('/');
 	};
+
+	this.shuffle = function (string) {
+		var a = string.split(""),
+			n = a.length;
+
+		for(var i = n - 1; i > 0; i--) {
```

Comment:
instead of first splitting the string and storing the new array and length then iterating and indexing, consider instead

```js
for (let s of string.split("")) {
    // 's' is now a letter in 'string'
}
```
But this still has a problem - splitting on `""` does **NOT** split on each character but actually between every UTF-16 codepoint. So for certain encodings - and afaik for things like region names we have no encoding restrictions - this can mangle the output. Instead, you could use the `[...someString]` spread operation, or [`Array.from`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from)

But, instead of iterating at all, consider the following:
```js
this.shuffle = (string) => {
    return [...string].sort(
        (a, b) => {
            return Math.floor(Math.random() * 3) - 1;
        }
    );
}
```

Reply:
this is not my area of expertise. the first one was a stackoverflow special.

yours is giving me weird output (commas):

commonFunctions.shuffle('abcdefghijklmonpqrstuvwxyz')
=> a,b,c,r,d,e,f,o,x,u,m,g,z,h,i,n,j,l,p,k,s,q,v,w,y,t

Reply:
oh i guess just convert output to a string..

Reply:
Oh, right. To get a string you'd want:
```javascript
this.shuffle = (string) => {
    return [...string].sort(
        (a, b) => {
            return Math.floor(Math.random() * 3) - 1;
        }
    ).join("");
}
```